beacon_node/beacon_chain/tests/store_tests.rs

```diff
@@ -373,12 +379,59 @@ fn delete_blocks_and_states() {
 
     let fork_blocks = 2 * E::slots_per_epoch();
 
-    let (honest_head, faulty_head) = harness.generate_two_forks_by_skipping_a_block(
-        &honest_validators,
-        &faulty_validators,
-        fork_blocks as usize,
-        fork_blocks as usize,
-    );
+    let mut slot = harness.get_current_slot() + 1;
+    let slot_u64: u64 = slot.into();
+
+    let fork1_slots: Vec<Slot> = (slot_u64..(slot_u64 + fork_blocks))
+        .map(Into::into)
+        .collect();
+    let fork2_slots: Vec<Slot> = (slot_u64 + 1..(slot_u64 + 1 + fork_blocks))
+        .map(Into::into)
+        .collect();
+
+    let mut fork1_state = harness.get_current_state();
+    let mut fork2_state = fork1_state.clone();
+
+    loop {
+        let epoch = slot.epoch(E::slots_per_epoch());
+        let epoch_start_slot: u64 = epoch.start_slot(E::slots_per_epoch()).into();
+        let epoch_end_slot: u64 = epoch.end_slot(E::slots_per_epoch()).into();
+        let this_epoch_slots: HashSet<Slot> =
+            (epoch_start_slot..epoch_end_slot).map(Into::into).collect();
+
+        let this_epoch_fork1_slots: Vec<Slot> = fork1_slots
+            .iter()
+            .filter(|slot| this_epoch_slots.contains(slot))
+            .copied()
+            .collect();
+        let this_epoch_fork2_slots: Vec<Slot> = fork2_slots
+            .iter()
+            .filter(|slot| this_epoch_slots.contains(slot))
+            .copied()
+            .collect();
+
+        if this_epoch_fork1_slots.is_empty() && this_epoch_fork2_slots.is_empty() {
+            break;
+        }
+
+        let (_, _, honest_head_, fork1_state_) = harness.add_attested_blocks_at_slots(
+            fork1_state,
+            &this_epoch_fork1_slots,
+            &honest_validators,
+        );
+        honest_head = honest_head_.into();
+        fork1_state = fork1_state_;
+
+        let (_, _, faulty_head_, fork2_state_) = harness.add_attested_blocks_at_slots(
+            fork2_state,
+            &this_epoch_fork2_slots,
+            &faulty_validators,
+        );
+        faulty_head = faulty_head_.into();
+        fork2_state = fork2_state_;
+
+        slot = epoch.end_slot(E::slots_per_epoch()) + 1;
+    }
```

Comment:
I had a go at abstracting this pattern (or something similar to it) into `add_blocks_on_multiple_chains`, which you can see an example of in the updated `pruning_test`. Would you mind refactoring this block in terms of the new primitive, and then we can merge?

Reply:
I was thinking of abstracting it that way too.  Just wanted to have something working first, then clean up.

Reply:
I did appreciate that! I was a bit dubious of whether it would work, but it worked almost as soon as I ran it :O